agent/rpc/peering/stream_test.go

```diff
@@ -795,9 +795,28 @@ func TestStreamResources_Server_ServiceUpdates(t *testing.T) {
 }
 
 type testStreamBackend struct {
-	pub    state.EventPublisher
-	store  *state.Store
-	leader func() bool
+	pub               state.EventPublisher
+	store             *state.Store
+	leader            func() bool
+	leadershipMonitor *leadershipMonitor
+}
+
+var _ LeadershipMonitor = (*leadershipMonitor)(nil)
+
+type leadershipMonitor struct {
```

Comment:
we could actually stub test the behavior of `trackLeadership` but then we would just test that a accessor/ mutator pair works under lock -- so there's not a lot of value there imo; that's why I wanted to test this in actuality at the consul server level